f3ast/stream_builder.py

```diff
@@ -128,3 +127,35 @@ def split_dwells(dwells, max_dwt):
         dwells_reduced = dwells.copy()
         dwells_reduced[:, 0] = dwells_reduced[:, 0] / n_splits
         return [dwells_reduced for i in range(n_splits)]
+
+
+def stream_exp_correction_with_z(struct, settings, GR0=40e-3, doubling_length=500., sigma=4.2):
```

Comment:
Looking at this function, I feel like this shouldn't be a standalone function. This should be a subclass of `Model` same as `RRLModel` and `DDModel`.
What you need to do is create a new class that inherits from `RRLModel` and modify `get_proximity_matrix` so that it takes into account the z height and reduces the growth rate exponentially with z. It's fairly easy to do and then it will naturally fit within the framework. I designed it so that you can easily plug in different models of how you are growing in this way.

I think this will be key to understand so that you can easily adjust the growth model depending on the conditions, so feel free to contact me if you want to discuss this further.